host/xtest/regression_1000.c

```diff
@@ -2687,3 +2694,361 @@ static void xtest_tee_test_1036(ADBG_Case_t *c)
+	st = RSA_verify_PKCS1_PSS_mgf1(att_key, digest, EVP_sha256(),
+				       EVP_sha256(), decr, salt_len);
+	ADBG_EXPECT_COMPARE_SIGNED(c, st, >, 0);
+}
+
+static void free_att_key(void)
+{
+	RSA_free(att_key);
+	att_key = NULL;
+}
+
+static void set_att_key(ADBG_Case_t *c, uint8_t *e, size_t e_sz, uint8_t *n,
+			size_t n_sz)
+{
+	BIGNUM *bn_e = NULL;
+	BIGNUM *bn_n = NULL;
+	int st = 0;
+
+	att_key_size = n_sz;
+	att_key = RSA_new();
+	if (!ADBG_EXPECT_NOT_NULL(c, att_key))
+		return;
+
+	bn_e = BN_bin2bn(e, e_sz, NULL);
+	if (!ADBG_EXPECT_NOT_NULL(c, bn_e))
+		goto err;
+	bn_n = BN_bin2bn(n, n_sz, NULL);
+	if (!ADBG_EXPECT_NOT_NULL(c, bn_n))
+		goto err;
+
+	st = RSA_set0_key(att_key, bn_n, bn_e, BN_new());
+	if (!ADBG_EXPECT_COMPARE_SIGNED(c, st, !=, 0))
+		goto err;
+	return;
+err:
+		free_att_key();
+}
+#else
+#define check_signature(...)
+#define set_att_key(...)
+#define free_att_key()
+#endif
+
+/*
+ * Verification of the output of the attestation PTA
+ * - (If hash != NULL) check buf contains the expected hash
+ * - (If OpenSSL is available) Check that the signature is valid
+ */
+static void check_measurement(ADBG_Case_t *c, uint8_t *nonce, size_t nonce_size,
+			      uint8_t *hash, uint8_t *buf)
+{
+	if (hash)
+		ADBG_EXPECT_BUFFER(c, hash, 32, buf, 32);
+
+	check_signature(c, nonce, nonce_size, buf);
+}
+
+/* Invoke attestation PTA to return the public key */
+static void get_att_public_key(ADBG_Case_t *c)
+{
+	uint8_t n[ATT_MAX_KEYSZ / 8] = { };
+	uint8_t e[3] = { };  /* We know e == 65537... */
+	size_t n_sz = sizeof(n);
+	size_t e_sz = sizeof(e);
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	TEEC_UUID att_uuid = PTA_ATTESTATION_UUID;
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	TEEC_Session session = { };
+	uint32_t ret_orig = 0;
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_OUTPUT,
+					 TEEC_MEMREF_TEMP_OUTPUT,
+					 TEEC_VALUE_OUTPUT, TEEC_NONE);
+	op.params[0].tmpref.buffer = e;
+	op.params[0].tmpref.size = e_sz;
+	op.params[1].tmpref.buffer = n;
+	op.params[1].tmpref.size = n_sz;
+
+	res = xtest_teec_open_session(&session, &att_uuid, NULL, &ret_orig);
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, res))
+		return;
+
+	res = TEEC_InvokeCommand(&session, PTA_ATTESTATION_GET_PUBKEY, &op,
+				 &ret_orig);
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, res) ||
+	    !ADBG_EXPECT_COMPARE_UNSIGNED(c, op.params[2].value.a, ==,
+				     TEE_ALG_RSASSA_PKCS1_PSS_MGF1_SHA256))
+		goto out;
+
+	e_sz = op.params[0].tmpref.size;
+	n_sz = op.params[1].tmpref.size;
+	set_att_key(c, e, e_sz, n, n_sz);
+out:
+	TEEC_CloseSession(&session);
+
+}
+
+/* Invoke attestation PTA to hash the TEE binary */
+static void attestation_tee(ADBG_Case_t *c)
+{
+	uint8_t measurement[TEE_SHA256_HASH_SIZE + ATT_MAX_KEYSZ / 8] = { 0 };
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint8_t nonce[4] = { 0x12, 0x34, 0x56, 0x78 };
+	TEEC_UUID att_uuid = PTA_ATTESTATION_UUID;
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	TEEC_Session session = { 0 };
```

Comment:
s/{ 0 }/{ }/

ditto at lines 2870 and 2932.

Reply:
Right and a few others even... fixed